File: zone-server/src/main/kotlin/net/bestia/zone/ecs/respawn/RespawnCMSG.kt
package net.bestia.zone.ecs.respawn

import net.bestia.bnet.proto.RespawnCmsgProto
import net.bestia.zone.message.CMSG

/**
 * Client asks for its currently active entity to be revived at its save point.
 */
data class RespawnCMSG(
  override val playerId: Long
) : CMSG {

  companion object {
    fun fromBnet(
      accountId: Long,
      @Suppress("UNUSED_PARAMETER") proto: RespawnCmsgProto.RespawnCMSG
    ): RespawnCMSG {
      return RespawnCMSG(playerId = accountId)
    }
  }
}

File: bestia-client/src/Bnet/Message/System/RespawnCMSG.cs
using Bnet;

namespace BestiaBehemothClient.Bnet.Message.System
{
  /// <summary>
  /// Asks the server to revive the currently active entity at its save point. Carries nothing: which
  /// entity acts comes from the session and where it belongs from the server's own records.
  /// </summary>
  public partial class RespawnCMSG : ICMSG
  {
    public override Envelope ToEnvelope()
    {
      return new Envelope
      {
        Respawn = new global::Bnet.RespawnCMSG()
      };
    }
  }
}
